tools/c7n_azure/c7n_azure/utils.py

```diff
@@ -473,8 +473,14 @@ def __str__(self):
             return self.str_value
 
     @staticmethod
-    def duration_from_period_and_units(period, retention_period_unit):
+    def iso8601_duration_from_period_and_units(period, retention_period_unit):
```

Comment:
nit: I think `iso8601_duration` is enough. `period` and `unit` is a parameter names. Totally up to you